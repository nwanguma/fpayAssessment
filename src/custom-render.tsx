import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const Wrapper: React.FC = ({ children }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
