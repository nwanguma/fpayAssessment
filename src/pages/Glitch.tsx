import { Link } from "react-router-dom";

import ErrorTemplate from "../templates/ErrorTemplate";

import notFound from "../assets/images/503.png";

const Glitch: React.FC = () => {
  return (
    <ErrorTemplate image={notFound}>
      <div className="container">
        <h2 className="text-[72px] leading-[90px] text-grey-large font-black">
          500
        </h2>
        <h3 className="mt-1 md:mt-3 text-[24px] leading-[30px] text-blue-black font-bold">
          There’s been a glitch…
        </h3>
        <p className="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text">
          An error has occurred and we’re working to fix the problem! We’ll be
          up and running shortly
        </p>
        <Link
          to="/"
          className="flex items-center mx-auto md:-mx-0 justify-center leading-[40px] h-[40px] w-[172px] bg-blue-tab text-white-main btn mt-8"
        >
          Refresh Page
        </Link>
      </div>
    </ErrorTemplate>
  );
};

export default Glitch;
