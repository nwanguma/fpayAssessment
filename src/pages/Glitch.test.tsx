import React from "react";
import { render, screen } from "../custom-render";

import Glitch from "./Glitch";

describe("Glitch", () => {
  test("<Glitch /> matches snapshot", () => {
    const component = render(<Glitch />);

    expect(component.container).toMatchSnapshot();
  });

  test("It renders children correctly", () => {
    const App = render(<Glitch />);

    expect(screen.getByTestId("container")).toBeInTheDocument();
  });

  test("It renders the correct text", () => {
    render(<Glitch />);

    expect(screen.getByText("There’s been a glitch…")).toBeInTheDocument();
  });

  test("It renders the correct link", () => {
    render(<Glitch />);

    expect(screen.getByText("Refresh Page").closest("a")).toHaveAttribute(
      "href",
      "/"
    );
  });
});

export {};
