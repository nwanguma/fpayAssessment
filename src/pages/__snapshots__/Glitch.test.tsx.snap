// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glitch <Glitch /> matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="wrapper"
  >
    <header
      class="header--payout w-[100%] md:w-[90%] mx-auto flex items-center py-6 justify-between"
    >
      <a
        href="/"
      >
        <img
          alt="fliqpay"
          src="logo.svg"
        />
      </a>
    </header>
    <main
      class="container min-h-[80vh] min-w-full flex items-center"
    >
      <div
        class="container flex flex-col-reverse  md:flex-row justify-center md:justify-between items-center max-w-[400px] md:text-left md:max-w-[80%] mx-auto"
      >
        <div
          class="container mt-3 flex-item w-[90%] text-center md:text-left"
        >
          <div
            class="container"
            data-testid="container"
          >
            <h2
              class="text-[72px] leading-[90px] text-grey-large font-black"
            >
              500
            </h2>
            <h3
              class="mt-1 md:mt-3 text-[24px] leading-[30px] text-blue-black font-bold"
            >
              There’s been a glitch…
            </h3>
            <p
              class="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text"
            >
              An error has occurred and we’re working to fix the problem! We’ll be up and running shortly
            </p>
            <a
              class="flex items-center mx-auto md:-mx-0 justify-center leading-[40px] h-[40px] w-[172px] bg-blue-tab text-white-main btn mt-8"
              href="/"
            >
              Refresh Page
            </a>
          </div>
        </div>
        <div
          class="container mx-auto md:mx-0 w-[250px] h-[150px] md:w-[350px] md:h-[250px]"
        >
          <img
            alt="error pic"
            class="w-[100%] h-[100%]"
            src="503.png"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
